Source/WebCore/html/canvas/CanvasRenderingContext2DBase.cpp

```diff
@@ -2384,14 +2388,26 @@ const Vector<CanvasRenderingContext2DBase::State, 1>& CanvasRenderingContext2DBa
 
 GraphicsContext* CanvasRenderingContext2DBase::drawingContext() const
 {
-    auto* context = canvasBase().drawingContext();
-    if (!context)
+    auto* buffer = canvasBase().buffer();
+    if (!buffer)
         return nullptr;
-
+    auto& context = buffer->context();
     if (UNLIKELY(m_targetSwitcher))
-        return m_targetSwitcher->drawingContext(*context);
+        return m_targetSwitcher->drawingContext(context);
+    return &context;
+}
 
-    return context;
+GraphicsContext* CanvasRenderingContext2DBase::existingDrawingContext() const
+{
+    if (!canvasBase().hasCreatedImageBuffer())
+        return nullptr;
+    return drawingContext();
+}
+
+AffineTransform CanvasRenderingContext2DBase::baseTransform() const
+{
+    ASSERT(canvasBase().hasCreatedImageBuffer());
+    return canvasBase().buffer()->baseTransform();
```

Comment:
This is a little bit confusing. If `CanvasBase` owns the `ImageBuffer` why does not it return the `baseTransform` also?

Reply:
Because this is work towards making CanvasBase not own the ImageBuffer.

Reply:
Please add a FIXME comment saying who should return the `baseTransform` if `CanvasBase` does not own the `ImageBuffer`